waltid-libraries/waltid-crypto/src/jvmMain/kotlin/id/walt/crypto/keys/JWKKeyGenerator.jvm.kt

```diff
@@ -39,26 +42,41 @@ object JvmJWKKeyCreator : JWKKeyCreator {
         return JWKKey(jwk)
     }
 
-    override suspend fun importRawPublicKey(type: KeyType, rawPublicKey: ByteArray, metadata: JwkKeyMeta?): Key = JWKKey(
-        when (type) {
-            KeyType.Ed25519 -> OctetKeyPair.Builder(Curve.Ed25519, Base64URL.encode(rawPublicKey)).build()
-            KeyType.secp256k1 -> ecRawToJwk(rawPublicKey, Curve.SECP256K1)
-            KeyType.secp256r1 -> ecRawToJwk(rawPublicKey, Curve.P_256)
-            KeyType.RSA -> rawRsaToJwk(rawPublicKey)
-        }
-    )
+    override suspend fun importRawPublicKey(type: KeyType, rawPublicKey: ByteArray, metadata: JwkKeyMeta?): Key =
+        JWKKey(
+            when (type) {
+                KeyType.Ed25519 -> OctetKeyPair.Builder(Curve.Ed25519, Base64URL.encode(rawPublicKey)).build()
+                KeyType.secp256k1 -> ecRawToJwk(rawPublicKey, Curve.SECP256K1)
+                KeyType.secp256r1 -> ecRawToJwk(rawPublicKey, Curve.P_256)
+                KeyType.RSA -> rawRsaToJwk(rawPublicKey)
+            }
+        )
 
     override suspend fun importJWK(jwk: String): Result<JWKKey> =
         runCatching { JWKKey(JWK.parse(jwk)) }
 
     override suspend fun importPEM(pem: String): Result<JWKKey> =
         runCatching { JWKKey(JWK.parseFromPEMEncodedObjects(pem)) }
 
-    private fun ecRawToJwk(rawPublicKey: ByteArray, curve: Curve): JWK =
-        ECNamedCurveTable.getParameterSpec(curve.name).curve.decodePoint(rawPublicKey).let {
-            val x: ByteArray = it.xCoord.encoded
-            val y: ByteArray = it.yCoord.encoded
-            return ECKey.Builder(curve, Base64URL.encode(x), Base64URL.encode(y)).build()
+    private fun ecRawToJwk(rawPublicKey: ByteArray, curve: Curve): JWK {
+        val bcEC = ECNamedCurveTable.getParameterSpec(curve.name).curve
+        val pubKeyEcPoint = decodeRawPubKeyEcPoint(bcEC, rawPublicKey)
+        return ECKey.Builder(
+            curve,
+            Base64URL.encode(pubKeyEcPoint.rawXCoord.encoded),
+            Base64URL.encode(pubKeyEcPoint.rawYCoord.encoded),
+        ).build()
+    }
+
+    private fun decodeRawPubKeyEcPoint(bcEC: ECCurve, rawEncodedPoint: ByteArray): ECPoint =
```

Comment:
I want to add a note here just to make it super clear for review purposes. The reason why this try-catch block was added is so that Uniresolver test cases in the did library don't start randomly breaking. Note that this ASN1 based decoding process is the correct one, but for some reason that doesn't work for the uniresolver stuff. I don't know why this is the case, perhaps they made a mistake when encoding the did strings, but this is nevertheless a fact.

I didn't want to delete these test cases (although I could have, since did:key resolution does no longer depend on the uniresolver), so this point is up for discussion which can be addressed by a small update in the context of this PR.